File: linear_search.cpp
#include<bits/stdc++.h>
using namespace std;
int linearsearch(int arr[],int n,int key)
{
    for(int i=0;i<n;i++)
    {
        if(arr[i]==key)
        {
            return i;
        }
    }
    return -1;
}
int main()
{
    int n,k;
    cout<<"Enter the size of the array"<<endl;;
    cin>>n;
    int arr[n];
    cout<<"Enter the array elements"<<endl;
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    cout<<"Enter the element to be searched"<<endl;
    cin>>k;
    int pos=linearsearch(arr,n,k);
    if(pos==-1)
    {
        cout<<"Element not found"<<endl;
    }
    else
    {
        cout<<"Element found at index"<<" "<<pos<<endl;
    }
    return 0;
}

File: Java/Searching/LinearSearch.java
// This class contains the implementation of Linear Search.
public class LinearSearch {

    // The main method is the entry point of the Java application.
    public static void main(String[] args) {

        // An array of integers to search within.
        int[] nums = {23, 45, 1, 2, 8, 19, -3, 16, -11, 28};

        // The target element to search for in the array.
        int target = 19;

        // Calling the linearSearch method and storing the result in 'ans'.
        boolean ans = linearSearch(nums, target);

        // Printing whether the target was found or not.
        System.out.println("Target found: " + ans);
    }

    /**
     * This method performs linear search on the given array.
     * @param arr The array to search in.
     * @param target The element to search for.
     * @return true if the target is found, otherwise false.
     */
    static boolean linearSearch(int[] arr, int target) {

        // Loop through each element in the array.
        for (int element : arr) {

            // If the current element matches the target, return true.
            if (element == target) {
                return true;
            }
        }

        // If the target is not found after checking all elements, return false.
        return false;
    }
}
